src/toast/observation_data.py

```diff
@@ -439,7 +439,13 @@ def __eq__(self, other):
             return False
         if self.units != other.units:
             return False
-        if not np.allclose(self.data, other.data):
+        if self.dtype == np.dtype(np.float64):
+            if not np.allclose(self.data, other.data, rtol=1.0e-5, atol=1.0e-8):
+                return False
+        elif self.dtype == np.dtype(np.float32):
+            if not np.allclose(self.data, other.data, rtol=1.0e-3, atol=1.0e-5):
```

Comment:
This seems like a very relaxed comparison for float32.

Reply:
Good point, I was trying to be flexible for the case of verifying data written to disk with FLAC compression and then loaded back in.  But I should just do a custom comparison for that case.  Will implement it.